pkg/controller/instance/instance_controller.go

```diff
@@ -382,22 +386,40 @@ func GetOperatorVersion(instance *kudov1beta1.Instance, c client.Client) (ov *ku
 	return ov, nil
 }
 
-// paramsMap generates {{ Params.* }} map of keys and values which is later used during template rendering.
-func paramsMap(instance *kudov1beta1.Instance, operatorVersion *kudov1beta1.OperatorVersion) map[string]string {
-	params := make(map[string]string)
-
-	for k, v := range instance.Spec.Parameters {
-		params[k] = v
+func paramValue(v *string, t v1beta1.ParameterValueType) (r interface{}, err error) {
```

Comment:
feels odd that this is in `instancecontroller`

it doesn't have anything to do with an instance... shouldn't this be encapsulated somewhere else?  perhaps you have thought about it already ?

Reply:
I agree, this shouldn't be here. `paramsMap` was already there, that's why I added `paramsValue` here as well. I'll see that this is move somewhere else, probably a separate package.

Reply:
It's tightly coupled to `paramsMap`, so i'm not totally opposed to having it here. On the other hand it kind of resembles code that is used in `verify_render.go`, so this could probably moved somewhere else.

Reply:
much ❤️ @nfnt ... I completely empathize with it was already here...   lets not make it a blocker.   if it is a challenge to refactor could you create an issue to capture it and we will move on.